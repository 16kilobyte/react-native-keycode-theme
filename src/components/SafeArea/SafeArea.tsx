import React from 'react';
import { SafeAreaView } from 'react-navigation';
import {
  ThemedComponentProps,
  ThemeType,
  withStyles,
} from 'react-native-ui-kitten';

const SafeArea: React.FC<{} & ThemedComponentProps> = ({
  children,
  themedStyle,
}) => {
  return <SafeAreaView style={themedStyle.container}>{children}</SafeAreaView>;
};

export default withStyles(SafeArea, (theme: ThemeType) => ({
  container: {
    flex: 1,
    backgroundColor: theme['background-basic-color-1'],
  },
}));
