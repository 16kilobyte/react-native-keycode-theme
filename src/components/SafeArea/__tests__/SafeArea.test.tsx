import React from 'react';
import { View } from 'react-native';
import { create } from 'react-test-renderer';

import SafeArea from '../SafeArea';

jest.useFakeTimers();

describe('SafeArea', () => {
  it('matches snapshot', () => {
    const component = create(<SafeArea>Hello</SafeArea>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('displays correct component', () => {
    const child = 'Hello';

    const component = create(<SafeArea>Hello</SafeArea>);
    const instance = component.root;
    expect(instance.findByType(View).props.children).toBe(child);
  });
});
