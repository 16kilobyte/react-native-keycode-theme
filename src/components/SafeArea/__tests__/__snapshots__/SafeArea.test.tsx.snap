// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeArea matches snapshot 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  Hello
</View>
`;
